### PR TITLE
Disable copy constructor of connection

### PR DESCRIPTION
**Pull request by Mytherin**
CC @ywelsch

### CODE REVIEW

File: src/main/connection.cpp [hunk 1]
```diff
@@ -30,7 +30,21 @@ Connection::Connection(DatabaseInstance &database) : context(make_shared<ClientC
 Connection::Connection(DuckDB &database) : Connection(*database.instance) {
 }
 
+Connection::Connection(Connection &&other) noexcept {
+	std::swap(context, other.context);
+	std::swap(warning_cb, other.warning_cb);
```
Comment (ywelsch) [inline src/main/connection.cpp:35]: `warning_cb` does not seem to be used anywhere AFAICS (same goes for `SetWarningCallback`, which is unimplemented). Maybe remove that field / method?

**Review by ywelsch — APPROVED** [review]
Thanks, LGTM

**Comment by dufferzafar** [comment]
We were using the connection's copy contructor in our codebase using duckdb 0.9.1 so are facing errors when upgrading to 0.10.1

@Mytherin / @ywelsch I wanted to understand what has fundamentally changed wrt the connections that copies are not allowed now & were allowed before.

**Comment by Mytherin** [comment]
The copy constructor was never intended to be used - this was a bug fix. If the connection is duplicated `RemoveConnection` could be called multiple times for the same `ClientContext`, which would leave the database with a connection it did not know about. This could then cause problems as e.g. checkpoints could no longer correctly verify that all connections were paused.

**Comment by dufferzafar** [comment]
So I guess we can use a shared_ptr<Connection> if we ever want to pass a connection around.

**Comment by Mytherin** [comment]
Yes, that would work

**Pull request merged**